pyatv/core/scan.py

```diff
@@ -288,3 +311,171 @@ def _end_if_identifier_found(self, response: mdns.Response):
         return self.identifier and not self.identifier.isdisjoint(
             set(get_unique_identifiers(response))
         )
+
+
+def _extract_service_name(info: AsyncServiceInfo) -> str:
+    return info.name[: -(len(info.type) + 1)]
+
+
+def _device_info_name(info: AsyncServiceInfo) -> Optional[str]:
+    if info.type not in NAME_USED_FOR_DEVICE_INFO:
+        return None
+    short_name = _extract_service_name(info)
+    if info.type == RAOP_TYPE:
+        return raop_name_from_service_name(short_name)
+    if info.type == SLEEP_PROXY_TYPE:
+        return sleep_proxy_name_from_service_name(short_name)
+    return short_name
+
+
+def _first_non_link_local_or_v6_address(addresses: List[bytes]) -> Optional[str]:
+    """Return the first ipv6 or non-link local ipv4 address."""
+    for address in addresses:
+        ip_addr = ip_address(address)
+        if not ip_addr.is_link_local or ip_addr.version == 6:
+            return str(ip_addr)
+    return None
+
+
+class AsyncDeviceInfoServiceInfo(AsyncServiceInfo):
+    """A version of AsyncServiceInfo that does not expect addresses."""
+
+    @property
+    def _is_complete(self) -> bool:
+        """Check if ServiceInfo has all expected properties.
+
+        The _device-info._tcp.local. does not return an address
+        so do not wait for it.
+        """
+        return self.text is not None
+
+
+class ZeroconfScanner(BaseScanner):
+    """Service discovery using zeroconf.
+
+    A ServiceBrowser must be running for all the types we are browsing.
+    """
+
+    def __init__(
+        self,
+        aiozc: AsyncZeroconf,
+        hosts: Optional[List[IPv4Address]] = None,
+    ) -> None:
+        """Initialize a new scanner."""
+        super().__init__()
+        self.aiozc = aiozc
+        self.zeroconf = aiozc.zeroconf
+        self.hosts: Set[str] = set(str(host) for host in hosts) if hosts else set()
+
+    async def _async_services_by_addresses(
```

Comment:
Minor thing, but as pyatv is "fully async" there's really no need for the `_async` prefix to spell out the context. I don't care, but just to mention it (I don't use it anywhere else).

Reply:
96827d4fa480cdd707a8bae09dc4a54efc8abb6a